File: iosApp/iosApp/swap/SwapTruthSource.swift
//
//  SwapTruthSource.swift
//  iosApp
//
//  Created by jimmyt on 8/15/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation

/**
 A protocol that a structure or class must adopt in order to act as a single source of truth for swap-related data.
 */
protocol SwapTruthSource {
    /**
     A dictionary mapping swap IDs (as `UUID`s) to `Swap`s, which is the single source of truth for all swap-related data.
     */
    var swaps: [UUID: Swap] { get set }
}

File: androidApp/src/main/java/com/commuto/interfacemobile/android/swap/SwapTruthSource.kt
package com.commuto.interfacemobile.android.swap

import androidx.compose.runtime.snapshots.SnapshotStateMap
import java.util.*

/**
 * An interface that a class must implement in order to act as a single source of truth for swap-related data.
 * @property swaps A [SnapshotStateMap] mapping [UUID]s to [Swap]s that should act as a single source of truth for all
 * swap-related data.
 */
interface SwapTruthSource {
    var swaps: SnapshotStateMap<UUID, Swap>

    /**
     * Should add a new [Swap] to [swaps].
     *
     * @param swap The new [Swap] that should be added to [swaps].
     */
    fun addSwap(swap: Swap)
}